internal/pyxis/pyxis.go

```diff
@@ -0,0 +1,80 @@
+package pyxis
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/shurcooL/graphql"
+)
+
+const (
+	DefaultPyxisHost = "catalog.redhat.com/api/containers"
+)
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
```

Comment:
There's probably an opportunity to simplify this code here. I don't see a good use case for this interface in this code & in this context. Especially given we type assert into an http.Client and have to handle an error in that process.

Reply:
I removed this and just use an `http.Client` directly.